TestSuite/repo.py

```diff
@@ -79,10 +80,23 @@ def __init__(self, tmpdir: Path):
                 "Wizards": [],
             }
         )
+        if as_git_repo:
+            GitUtil.REPO_CLS.init(self.path)
+            self.git_util = GitUtil(Path(self.path))
+        else:
+            self.git_util = None  # type: ignore[assignment]
 
     def __del__(self):
         shutil.rmtree(self.path, ignore_errors=True)
 
+    @classmethod
+    def init_as_git_repo(cls, tmpdir: Path) -> "Repo":
+        content_repo = cls(tmpdir, as_git_repo=True)
+        # commit all files
+        content_repo.git_util.commit_files("Initial Commit")
+        content_repo.git_util.repo.create_head("master").checkout()
+        return content_repo
```

Comment:
why both this and the `if` in `__init__`?

Reply:
updated the implementation